### ISSUE TITLE
Phase 2: Neural Biome Blending — Elevation-Dependent Biomes via MLP

### ISSUE THREAD

**Issue by plagness** [plagness/Metalcraft#12]
## Summary

Replace the Gaussian RBF biome classification system (`getBiomeWeights()`) with a small MLP that incorporates elevation, water proximity, and latitude for more natural biome transitions.

## Background

Current biome system in `VoxelTerrainGen.metal` (lines 185-216):
- 6 fixed 2D points in (temperature, humidity) space
- Exponential distance weighting: `exp(-dist * 7.0)`
- No elevation dependency (mountain tundra can appear at sea level)
- No awareness of water proximity or latitude

## Network Architecture

```
Input:  6 float — temperature, humidity, continental_noise, elevation,
                  distance_to_water, latitude_proxy
Hidden: 2 × 16 neurons, ReLU
Output: 6 float (softmax) — weights for 6 biomes
Parameters: 6×16 + 16 + 16×16 + 16 + 16×6 + 6 = 502 weights (~1 KB FP16)
```

## Implementation Plan

- [ ] Add `neuralBiomeWeights()` MLP in `VoxelTerrainGen.metal`
- [ ] Compute additional inputs (elevation, water distance, latitude) from existing noise
- [ ] Softmax output ensures weights sum to 1.0
- [ ] Share weights buffer with Phase 1 (combined: 1,818 params = ~3.6 KB)
- [ ] Generate training data from current system + manual corrections
- [ ] Train and export model

## Verification
- Biome map visualization at different elevations
- Check that biomes transition smoothly at borders
- Mountain tundra only above Y=80, coastal biomes near sea level
